localboot/main.go

```diff
@@ -110,13 +87,116 @@ func main() {
 		}
 	}
 	log.Print("No boot configuration succeeded")
-
-	// if we are here, booting failed, so let's clean things up by closing all
-	// open file descriptors and unmounting the devices that we mounted
-	for _, cfg := range bootconfigs {
-		cfg.Close()
-	}
+	// clean up
 	for _, mountpoint := range mounted {
 		syscall.Unmount(mountpoint.Path, syscall.MNT_DETACH)
 	}
+	return nil
+}
+
+// BootPathMode tries to boot a kernel in PATH mode. This means:
+// * look for a partition with the given GUID and mount it
+// * look for the kernel and initramfs in the provided locations
+// * boot the kernel with the provided command line
+//
+// The first parameter, `devices`, is a list of `storage.BlockDev` objects to
+// look for the given GUID. The second parameter, `guid`, is the partition GUID
+// to look for. The third, `dryrun`, will not actually boot the found
+// configuration if set to true.
+func BootPathMode(devices []storage.BlockDev, baseMountpoint string, guid string, dryrun bool) error {
```

Comment:
See above, params have changed names, and baseMountpoint hides the outer var again